rotkehlchen/chain/evm/manager.py

```diff
@@ -0,0 +1,1008 @@
+    Dict,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+)
+from urllib.parse import urlparse
+
+import requests
+from ens import ENS
+from eth_abi.exceptions import InsufficientDataBytes
+from eth_typing import BlockNumber
+from web3 import HTTPProvider, Web3
+from web3._utils.abi import get_abi_output_types
+from web3._utils.contracts import find_matching_event_abi
+from web3._utils.filters import construct_event_filter_params
+from web3.datastructures import MutableAttributeDict
+from web3.exceptions import (
+    BadFunctionCallOutput,
+    BadResponseFormat,
+    BlockNotFound,
+    TransactionNotFound,
+)
+from web3.types import BlockIdentifier, FilterParams
+
+from rotkehlchen.chain.constants import DEFAULT_EVM_RPC_TIMEOUT
+from rotkehlchen.chain.ethereum.graph import Graph
+from rotkehlchen.chain.ethereum.types import NodeName, WeightedNode
+from rotkehlchen.chain.ethereum.utils import MULTICALL_CHUNKS
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.constants import ONE
+from rotkehlchen.constants.ethereum import ERC20TOKEN_ABI, UNIV1_LP_ABI
+from rotkehlchen.constants.resolver import ChainID
+from rotkehlchen.errors.misc import BlockchainQueryError, RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.externalapis.etherscan import Etherscan
+from rotkehlchen.fval import FVal
+from rotkehlchen.greenlets import GreenletManager
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import (
+    deserialize_evm_address,
+    deserialize_evm_transaction,
+    deserialize_int_from_hex,
+)
+from rotkehlchen.serialization.serialize import process_result
+from rotkehlchen.types import (
+    ChecksumEvmAddress,
+    EvmTransaction,
+    EVMTxHash,
+    SupportedBlockchain,
+    Timestamp,
+)
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.misc import from_wei, get_chunks, hex_or_bytes_to_str
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _is_synchronized(current_block: int, latest_block: int) -> Tuple[bool, str]:
+    """ Validate that the evm node is synchronized
+            within 20 blocks of latest block
+
+        Returns a tuple (results, message)
+            - result: Boolean for confirmation of synchronized
+            - message: A message containing information on what the status is.
+    """
+    message = ''
+    if current_block < (latest_block - 20):
+        message = (
+            f'Found evm node but it is out of sync. {current_block} / '
+            f'{latest_block}. Will use etherscan.'
+        )
+        log.warning(message)
+        return False, message
+
+    return True, message
+
+
+WEB3_LOGQUERY_BLOCK_RANGE = 250000
+
+
+class EvmManager:
+    """EvmManager defines a basic implementation for most of EVM chains."""
```

Comment:
Since this one is not supposed to be instanciated I guess using `ABCMeta` is better with `abstractmethod`

Reply:
Done!